otherlibs/stdune/predicate.mli

```diff
@@ -0,0 +1,13 @@
+(** Predicates are functions of type 'a -> bool *)
+
+type 'a t = 'a -> bool
```

Comment:
Can we make this type abstract? Exposing it doesn't seem necessary as we already have [create] and [test] functions. It may also make our lifes slighly easier if we decide to defunctionalize this type.

Reply:
Sure, I don't mind making it abstract. But note that we wouldn't make it structural here. Stdune is for general purpose things and a structural predicate is specialized enough that it would have to be in the engine itself.

Reply:
Sounds good to me

Reply:
Done.